rush.json

```diff
@@ -485,6 +485,10 @@
     {
       "packageName": "@microsoft/rush-lib",
       "projectFolder": "apps/rush-lib",
+      "cyclicDependencyProjects": [
+        "@rushstack/rush-amazon-s3-build-cache-plugin",
+        "@rushstack/rush-azure-storage-build-cache-plugin"
```

Comment:
@chengcyber  This setting does not really eliminate the cycle. Its effect is to prevent `rush build` from building these projects /before/ `rush-lib` is built.  It will succeed if there are left over files from a previous build, but it will fail after `rush purge` or a clean `git clone`.

How I understand the problem:

1. `rush-lib` has a strong dependency on the builtin plugins (AWS and Azure) because (1) it needs to `require()` them for testing and (2) its **package.json** must have a dependency during `npm publish`.
2. All plugins (not only builtin plugins) need Rush's .d.ts types to compile their code.  This is a `devDependency`.
3. The unit tests for all plugins (not only builtin plugins) will need Rush's .js files to construct `RushSession`. This is also a `devDependency`.

**Approach A:**  My original suggestion was to create a fake project called `rush-builtin-plugin-types` which exports a committed copy of `rush-lib.d.ts` only.  The builtin plugins can import from this instead of importing from `rush-lib`.  This works, but the tests fail because of issue 3.   We could redesign the unit tests so that they mock `RushSession` (which eliminates `require('rush-lib')` for Jest) -- but that seems too restrictive.  (What do you think?)

**Approach B:** Remove `cyclicDependencyProjects` from `rush-lib`, and instead add `"cyclicDependencyProjects": [ "@microsoft/rush-lib" ]` for the plugin projects, and also remove `workspace:*` in their **package.json**.  This way they install the last published release of Rush, which breaks the cycle.  It doesn't work because `RushSession` has not been published yet.  That will get solved after the first release👍, but it will reappear whenever a PR introduces a new API that is used by the plugin projects👎.  That might be a bad experience for other engineers who don't understand this tricky relationship.

**Approach C:**  Run `npm pack` in `apps/rush-lib` and add that to Git.  Then the plugin projects take a dependency on `file:../microsoft-rush-lib-5.56.0.tgz` instead of `workspace:*`.  This works, but is ugly.  We don't like opaque binaries in our `git log`.

**Approach D:**  Create a fake project called `rush-builtin-plugin-types` like in Approach A, but extract the unit tests into a separate project.  For example `rush-amazon-s3-build-cache-plugin-tests` depends on `rush-lib` and also on `rush-amazon-s3-build-cache-plugin`.  This way the plugin builds first using `rush-builtin-plugin-types/rush-lib.d.ts`, then `rush-lib` builds, then the plugin's unit tests are run last.  (This has the advantage of ensuring that `rush-amazon-s3-build-cache-plugin` cannot import `rush-lib` by accident.)

**Approach E:** Remove the plugins from `rush-lib/package.json` and somehow temporarily disable loading of builtin plugins.  Then a CI job (or a human process) will revert these changes immediately before `rush publish` is invoked.  This is closest to "correct" because we planned to eliminate the builtin plugins once the feature is stable.  But until that happens, it may be a bit awkward for us to debug the build cache.

What do you think?